lightning/src/ln/channelmanager.rs

```diff
@@ -12967,7 +12967,7 @@ where
 									// inbound edge of the payment's monitor has already claimed
 									// the HTLC) we skip trying to replay the claim.
 									let htlc_payment_hash: PaymentHash = payment_preimage.into();
-									if inbound_edge_balances.iter().any(|bal| {
+									if !inbound_edge_balances.iter().any(|bal| {
```

Comment:
This is still difficult to understand, IMO. I think it would be better to remove the `!` and define a local variable with a suitable name that is check next in an `if` statement rather than inlining it.

```rust
let some_condition = inbound_edge_balances.iter().any(|bal| /* */);
if !some_condition {
	return None;
}
```